manager/src/test/java/com/redhat/service/bridge/manager/BridgesServiceTest.java

```diff
@@ -193,4 +197,35 @@ public void getBridge() {
     public void getBridge_bridgeDoesNotExist() {
         assertThatExceptionOfType(ItemNotFoundException.class).isThrownBy(() -> bridgesService.getBridge("foo"));
     }
+
+    @Test
+    public void testDeleteBridge() {
+        BridgeRequest request = new BridgeRequest(TestConstants.DEFAULT_BRIDGE_NAME);
+        bridgesService.createBridge(TestConstants.DEFAULT_CUSTOMER_ID, request);
+
+        //Wait for Workers to complete
+        Bridge bridge = TestUtils.waitForBridgeToBeReady(bridgesService);
+
+        bridgesService.deleteBridge(bridge.getId(), bridge.getCustomerId());
+
+        Bridge retrievedBridge = bridgesService.getBridge(bridge.getId(), TestConstants.DEFAULT_CUSTOMER_ID);
+        assertThat(retrievedBridge.getStatus()).isEqualTo(ManagedResourceStatus.DELETED);
+    }
+
+    @Test
+    public void testDeleteBridge_whenStatusIsNotReady() {
+        Bridge bridge = createPersistBridge(ManagedResourceStatus.PROVISIONING);
+        assertThatExceptionOfType(BridgeLifecycleException.class).isThrownBy(() -> bridgesService.deleteBridge(bridge.getId(), bridge.getCustomerId()));
+    }
+
+    private Bridge createPersistBridge(ManagedResourceStatus status) {
```

Comment:
@vaibhavjainwiz Take a look at `Fixtures.createBridge()`. It should give you most of what you need here.

Reply:
done